packages/malloy-interfaces/src/nest_unions.ts

```diff
@@ -66,11 +106,65 @@ export function unnestUnions(obj: unknown, type: string): unknown {
         if (childType === undefined) {
           throw new Error(`Unknown field ${key} in ${type}`);
         }
-        result[key] = unnestUnions(obj[key], childType.type);
+        result[key] = convertFromThrift(obj[key], childType.type);
       }
       return result;
     } else {
       throw new Error(`Cannot unnest unions in an enum type ${type}`);
     }
   }
 }
+
+function getType(type: string) {
+  const typeDefinition = MALLOY_INTERFACE_TYPES[type];
+  if (typeDefinition === undefined) {
+    throw new Error(`Unknown Malloy interface type ${type}`);
+  }
+  return typeDefinition;
+}
+
+export function convertToThrift(obj: unknown, type: string): unknown {
+  if (obj === null || obj === undefined) {
+    return obj;
+  } else if (typeof obj === 'number') {
+    return obj;
+  } else if (typeof obj === 'string') {
+    if (type === 'string') return obj;
+    const typeDefinition = getType(type);
+    if (typeDefinition.type === 'enum') {
+      return typeDefinition.values.indexOf(obj) + 1;
```

Comment:
Are you able to grab the id that you define in the thrift? not just the position in the array. I think that'd be the correct way.

This works because right now you have:
```
enum NumberSubtype {
  INTEGER = 1,
  DECIMAL = 2,
}
```

what if you had:
```
enum NumberSubtype {
  INTEGER = 35,
  DECIMAL = 55,
}
```